src/transformers/models/d_fine/modular_d_fine_resnet.py

```diff
@@ -0,0 +1,414 @@
+            temp_in_channels = in_channels if i == 0 else middle_channels
+            if light_block:
+                block = DFineResNetConvLayerLight(
+                    in_channels=temp_in_channels,
+                    out_channels=middle_channels,
+                    kernel_size=kernel_size,
+                    use_learnable_affine_block=use_learnable_affine_block,
+                )
+            else:
+                block = DFineResNetConvLayer(
+                    in_channels=temp_in_channels,
+                    out_channels=middle_channels,
+                    kernel_size=kernel_size,
+                    use_learnable_affine_block=use_learnable_affine_block,
+                    stride=1,
+                )
+            self.layers.append(block)
+
+        # feature aggregation
+        total_chs = in_channels + layer_num * middle_channels
+        if aggregation == "se":
+            aggregation_squeeze_conv = DFineResNetConvLayer(
+                total_chs,
+                out_channels // 2,
+                kernel_size=1,
+                stride=1,
+                use_learnable_affine_block=use_learnable_affine_block,
+            )
+            aggregation_excitation_conv = DFineResNetConvLayer(
+                out_channels // 2,
+                out_channels,
+                kernel_size=1,
+                stride=1,
+                use_learnable_affine_block=use_learnable_affine_block,
+            )
+            self.aggregation = nn.Sequential(
+                aggregation_squeeze_conv,
+                aggregation_excitation_conv,
+            )
+        else:
+            aggregation_conv = DFineResNetConvLayer(
+                total_chs, out_channels, kernel_size=1, stride=1, use_learnable_affine_block=use_learnable_affine_block
+            )
+            att = DFineResNetEseModule(out_channels)
+            self.aggregation = nn.Sequential(
+                aggregation_conv,
+                att,
+            )
+        self.drop_path = nn.Dropout(drop_path) if drop_path else nn.Identity()
+
+    def forward(self, hidden_state: Tensor) -> Tensor:
+        identity = hidden_state
+        output = [hidden_state]
+        for layer in self.layers:
+            hidden_state = layer(hidden_state)
+            output.append(hidden_state)
+        hidden_state = torch.cat(output, dim=1)
+        hidden_state = self.aggregation(hidden_state)
+        if self.residual:
+            hidden_state = self.drop_path(hidden_state) + identity
+        return hidden_state
+
+
+class DFineResNetStage(nn.Module):
+    def __init__(
+        self,
+        in_channels: int,
+        mid_channels: int,
+        out_channels: int,
+        block_num: int,
+        layer_num: int,
+        downsample: bool = True,
+        light_block: bool = False,
+        kernel_size: int = 3,
+        aggregation: str = "se",
+        drop_path: float = 0.0,
+        use_learnable_affine_block: bool = False,
+    ):
```

Comment:
btw can we initialize it from config with the given stage index as an additional argument?